lib/ansible/modules/cloud/amazon/rds_param_group.py

```diff
@@ -175,172 +168,231 @@ def __str__(self):
+                             exception=traceback.format_exc(),
+                             **camel_dict_to_snake_dict(e.response))
+        except botocore.exceptions.ParamValidationError as e:
+            # Usually a tag value has been passed as an int or bool, needs to be a string
+            # The AWS exception message is reasonably ok for this purpose
+            module.fail_json(msg="Couldn't add tags to parameter group: %s." % str(e),
+                             exception=traceback.format_exc())
+    if to_delete and module.params['purge_tags']:
+        try:
+            connection.remove_tags_from_resource(ResourceName=group['DBParameterGroupArn'],
+                                                 TagKeys=to_delete)
+            changed = True
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg="Couldn't remove tags from parameter group: %s" % str(e),
+                             exception=traceback.format_exc(),
+                             **camel_dict_to_snake_dict(e.response))
+    return changed
+
+
+def ensure_present(module, connection):
+    groupname = module.params['name']
+    tags = module.params.get('tags')
+    changed = False
+    errors = []
+    try:
+        response = connection.describe_db_parameter_groups(DBParameterGroupName=groupname)
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == 'DBParameterGroupNotFound':
+            response = None
+        else:
+            module.fail_json(msg="Couldn't access parameter group information: %s" % str(e),
+                             exception=traceback.format_exc(),
+                             **camel_dict_to_snake_dict(e.response))
+    if not response:
+        params = dict(DBParameterGroupName=groupname,
+                      DBParameterGroupFamily=module.params['engine'],
+                      Description=module.params['description'])
+        if tags:
+            params['Tags'] = ansible_dict_to_boto3_tag_list(tags)
+        try:
+            response = connection.create_db_parameter_group(**params)
+            changed = True
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg="Couldn't create parameter group: %s" % str(e),
+                             exception=traceback.format_exc(),
+                             **camel_dict_to_snake_dict(e.response))
+    else:
+        group = response['DBParameterGroups'][0]
+        if tags:
+            changed = update_tags(module, connection, group, tags)
 
-            if old_value != new_value:
-                if not param.is_modifiable:
-                    raise NotModifiableError('Parameter %s is not modifiable.' % key)
+    if module.params.get('params'):
+        params_changed, errors = update_parameters(module, connection)
+        changed = changed or params_changed
 
-                changed[key] = {'old': old_value, 'new': new_value}
```

Comment:
Changed is evaluated as true/false, but is a dict of whatever values have been changed and their before/after value. Here's a playbook and what it's doing on devel:
```
---
- hosts: localhost
  connection: local
  tasks:
    - name: create rds parameter group
      rds_param_group:
        name: "test-rds-param-group"
        description: "test-parameter-group"
        params:
          auto_increment_increment: "42K"
        engine: postgres9.5
        state: present
    - name: update parameter group
      rds_param_group:
        name: "test-rds-param-group"
        description: "test-parameter-group"
        params:
          auto_increment_increment: "42K"
          application_name: "test-param"
        engine: postgres9.5
        state: present
```
```
ok: [localhost] => {
    "changed": {},
    "failed": false,
    "invocation": {
        "module_args": {
            "aws_access_key": null,
            "aws_secret_key": null,
            "description": "test-parameter-group",
            "ec2_url": null,
            "engine": "postgres9.5",
            "immediate": null,
            "name": "test-rds-param-group",
            "params": {
                "auto_increment_increment": "42K"
            },
            "profile": "shertel",
            "region": "us-east-1",
            "security_token": null,
            "state": "present",
            "validate_certs": true
        }
    }
}

TASK [update parameter group] **********************************************************************************
task path: /Users/shertel/Workspace/ansible/my_playbooks/rds_param_group/try1.yml:25
Using module file /Users/shertel/Workspace/ansible/lib/ansible/modules/cloud/amazon/rds_param_group.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: shertel
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/shertel/.ansible/tmp/ansible-tmp-1500298806.78-269063698706556 `" && echo ansible-tmp-1500298806.78-269063698706556="` echo /Users/shertel/.ansible/tmp/ansible-tmp-1500298806.78-269063698706556 `" ) && sleep 0'
<127.0.0.1> PUT /var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/tmpUTbHaQ TO /Users/shertel/.ansible/tmp/ansible-tmp-1500298806.78-269063698706556/rds_param_group.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /Users/shertel/.ansible/tmp/ansible-tmp-1500298806.78-269063698706556/ /Users/shertel/.ansible/tmp/ansible-tmp-1500298806.78-269063698706556/rds_param_group.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/Users/shertel/Workspace/ansible/venv/python2.7/bin/python /Users/shertel/.ansible/tmp/ansible-tmp-1500298806.78-269063698706556/rds_param_group.py; rm -rf "/Users/shertel/.ansible/tmp/ansible-tmp-1500298806.78-269063698706556/" > /dev/null 2>&1 && sleep 0'
changed: [localhost] => {
    "changed": {
        "application_name": {
            "new": "test-param",
            "old": null
        }
    },
    "failed": false,
    "invocation": {
        "module_args": {
            "aws_access_key": null,
            "aws_secret_key": null,
            "description": "test-parameter-group",
            "ec2_url": null,
            "engine": "postgres9.5",
            "immediate": null,
            "name": "test-rds-param-group",
            "params": {
                "application_name": "test-param",
                "auto_increment_increment": "42K"
            },
            "profile": "shertel",
            "region": "us-east-1",
            "security_token": null,
            "state": "present",
            "validate_certs": true
        }
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *****************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0
```